### ISSUE TITLE
Crash MPRateLimitManager.m line 41

### ISSUE THREAD

**Issue by aechap02** [mopub/mopub-ios-sdk#269]
Seeing a crash occur, not consistently enough to reproduce it ourselves but seeing it appear in Crashlytics.

- [x] I am submitting a bug report for existing functionality
- [x] I visited https://developers.mopub.com/ and found no answer
- [x] I checked https://twittercommunity.com/c/advertiser-api/mopub to make sure that this issue has not already been filed
- [x] I checked to make sure that this issue has not already been filed

#### MoPub SDK Version: 
5.6.0

#### Device model and OS Version: 
iPhone (5s, 6, 6s, 6 Plus, 7, 8, X, etc), iPad (Mini, Mini 2, Mini 4, Pro, etc)
iOS 9 - iOS 12

#### Xcode Version:
XCode 10.1

#### Steps to reproduce the behavior:
We haven't been able reproduce this issue, but it is currently one of the issues with the highest number of crashes on our Fabric crashlytics dashboard. Effecting approximately 4-5% of users.

64% had the crash occur while the app was in the background, if that matters.

#### Evidence - Stack trace:
```
Fatal Exception: NSInvalidArgumentException
0  CoreFoundation                 0x1c1790518 __exceptionPreprocess
1  libobjc.A.dylib                0x1c096b9f8 objc_exception_throw
2  CoreFoundation                 0x1c1709f8c _CFArgv
3  CoreFoundation                 0x1c1685520 -[__NSDictionaryM setObject:forKeyedSubscript:]
4  MoPub                          0x102059e78 -[MPRateLimitManager setRateLimitTimerWithAdUnitId:milliseconds:reason:] (MPRateLimitManager.m:41)
5  MoPub                          0x10201b668 -[MPAdServerCommunicator didFinishLoadingWithData:] (MPAdServerCommunicator.m:239)
6  MoPub                          0x10201a624 __34-[MPAdServerCommunicator loadURL:]_block_invoke (MPAdServerCommunicator.m:107)
7  libdispatch.dylib              0x1c11d0a38 _dispatch_call_block_and_release
8  libdispatch.dylib              0x1c11d17d4 _dispatch_client_callout
9  libdispatch.dylib              0x1c11b19e4 _dispatch_main_queue_callback_4CF$VARIANT$armv81
10 CoreFoundation                 0x1c1721ec0 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__
11 CoreFoundation                 0x1c171cdf8 __CFRunLoopRun
12 CoreFoundation                 0x1c171c354 CFRunLoopRunSpecific
13 GraphicsServices               0x1c391c79c GSEventRunModal
14 UIKitCore                      0x1edbc5b68 UIApplicationMain
15 <App>                          0x100dcbee8 main (AppDelegate.swift:41)
16 libdyld.dylib                  0x1c11e28e0 start
```

**Comment by chauduyphanvu1** [comment]
@aechap02 - thanks for letting us know! I'll take a look into what might be causing this crash, and share details on this thread.

Vu Chau
MoPub SDK Team

**Comment by chauduyphanvu1** [comment]
@aechap02 - while we did not reproduce that exact crash, we managed to trigger something very close to it. The team has put in a fix that we believe should fix the issue on this report. It should be available in an upcoming SDK release.

**Comment by aechap02** [comment]
Thank you!

**Comment by andypoptacular** [comment]
Any idea then the fix for this might land? It's our number one crash so considering our options (e.g. rollback SDK, try our own fix/workaround.  Any update would be great appreciated.

**Comment by chauduyphanvu1** [comment]
@andypoptacular, it should go out with the 5.7.1 SDK release, which should be released soon. I'll get back to you if that's not the case.

**Comment by rsatrasala** [comment]
Fixed in 5.7.1
